PWGUD/Core/UPCJpsiCentralBarrelCorrHelper.h

```diff
@@ -0,0 +1,193 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+///
+/// \brief
+/// \author Sara Haidlova, sara.haidlova@cern.ch
+/// \since March 2024
+
+#ifndef PWGUD_CORE_UPCJPSICENTRALBARRELCORRHELPER_H_
+#define PWGUD_CORE_UPCJPSICENTRALBARRELCORRHELPER_H_
+
+#include <algorithm>
+#include "CommonConstants/MathConstants.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+using namespace std;
+
+enum ParticleType {
+  P_ELECTRON = 0,
+  P_MUON = 1,
+  P_PROTON = 2
+};
+
+template <typename T>
+int testPIDhypo(T trackPID)
+{
+  float nSigmaTPC[3];
+  nSigmaTPC[P_ELECTRON] = std::abs(trackPID.tpcNSigmaEl());
+  nSigmaTPC[P_MUON] = std::abs(trackPID.tpcNSigmaMu());
+  nSigmaTPC[P_PROTON] = std::abs(trackPID.tpcNSigmaPr());
+  int enumChoiceTPC = std::distance(std::begin(nSigmaTPC),
+                                    std::min_element(std::begin(nSigmaTPC), std::end(nSigmaTPC)));
+  if (trackPID.hasTPC()) {
+    return enumChoiceTPC;
+  } else {
+    return -1;
+  }
+}
+
+float* correlation(TLorentzVector* lv1, TLorentzVector* lv2, TLorentzVector* lv)
+{
+  TLorentzVector pa(1., 0., 0, 1);  // projectile
+  TLorentzVector pb(1., 0., 0, -1); // target
+
+  float* q = new float[3]; // to save values
+
+  // Accoplanarity angle
+  Float_t deltaPhi;
+  deltaPhi = lv1->Phi() - lv2->Phi();
+  float accoplCut = 1. - abs(deltaPhi) / o2::constants::math::PI;
+  // z
+  TLorentzVector z;
+  Float_t part1, part2;
+
+  // Dot product: v1*v2 = t1*t2-x1*x2-y1*y2-z1*z2
+
+  part1 = lv->Dot(pb);
+  part2 = lv->Dot(pa);
+
+  Float_t part3x = pa.X() * part1;
+  Float_t part3y = pa.Y() * part1;
+  Float_t part3z = pa.Z() * part1;
+  Float_t part3e = pa.T() * part1;
+
+  Float_t part4x = pb.X() * part2;
+  Float_t part4y = pb.Y() * part2;
+  Float_t part4z = pb.Z() * part2;
+  Float_t part4e = pb.T() * part2;
+
+  TLorentzVector part3(TVector3(part3x, part3y, part3z), part3e);
+  TLorentzVector part4(TVector3(part4x, part4y, part4z), part4e);
+
+  // Un-normalized Z
+  z = part3 - part4;
+
+  // Normalized z
+  Float_t normz = TMath::Sqrt(-z * z);
```

Comment:
```suggestion
  Float_t normz = std::sqrt(-z * z);
```
There is no benefit using ROOT functions instead of the standard C++ library.